### PR TITLE
Add RSS support for Blog entries

### PR DESCRIPTION
**Pull request by msurdi**
### ❓ Type of change

- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

I wanted to subscribe to your blog using an RSS feed reader and noticed there was no feed for the blog. Here's a quick implementation for adding it.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Julien-R44** [comment]
Cool thing to have! Thanks for the PR

Looks good to me. @thetutlage any thoughts ?

**Comment by RomainLanz** [comment]
Hey @msurdi! 👋🏻 

Thanks for this PR.

Since we are dumping the whole article content, the file will probably grow big.

Wouldn't it be better only to display the title and a summary?

**Comment by msurdi** [comment]
Hi @RomainLanz! Thanks a lot for taking some time to review this.

I think it's nice to have the full article, so that you can read the posts without leaving your news reader. But I agree the size of this can get quickly out of hand.

I've just pushed a commit where I instead limited the number of posts to the latest 20 available. Would that work?

I'm happy to truncate the content if you still think it's necessary and there's no way around that.

Thanks!

**Comment by thetutlage** [comment]
I think its fine to have all the contents for now, since we don't have many posts. Later we can see if we want to something different once we start to hit any sort of limits

**Comment by thetutlage** [comment]
We will also have to update the `build:static` command and write the `blog.xml` file on the disk, since the production site is a static output. https://github.com/adonisjs/adonisjs.com/blob/main/commands/build_static.ts

**Comment by RomainLanz** [comment]
> I think its fine to have all the contents for now, since we don't have many posts.

Agreed.

One question though: do most RSS readers support JavaScript? Since we might have custom components in those articles, they could potentially break in RSS readers if JavaScript isn’t handled.

**Comment by thetutlage** [comment]
To be honest, I have not used any RSS readers recently. So yeah, we should give a test run once and see what the ideal experience will be

**Comment by msurdi** [comment]
@thetutlage I've just added the static generation step for the feed.

Let me know if there is anything else I can do here.

Thanks 🙏

File: resources/views/feeds/blog.edge [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<rss xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:content="http://purl.org/rss/1.0/modules/content/" xmlns:atom="http://www.w3.org/2005/Atom" version="2.0">
+  <channel>
+    <title>Adonis Framework Blog</title>
```
Comment (RomainLanz) [inline resources/views/feeds/blog.edge:4]:
```suggestion
    <title>AdonisJS Framework Blog</title>
```

File: resources/views/feeds/blog.edge [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    <link>https://adonisjs.com/</link>
+    <description>All the latest Adonis Framework news.</description>
```
Comment (RomainLanz) [inline resources/views/feeds/blog.edge:6]:
```suggestion
    <description>All the latest AdonisJS Framework news.</description>
```

File: resources/views/feeds/blog.edge [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    <lastBuildDate>{{new Date()}}</lastBuildDate>
+    <docs>https://validator.w3.org/feed/docs/rss2.html</docs>
+    <language>en</language>
+    <image>
+    <title>Adonis Framework Blog</title>
```
Comment (RomainLanz) [inline resources/views/feeds/blog.edge:11]:
```suggestion
    <title>AdonisJS Framework Blog</title>
```

**Review by RomainLanz — CHANGES_REQUESTED** [review]

**Comment by msurdi** [comment]
All suggested changes applied.

**Review by RomainLanz — APPROVED** [review]
LGTM

**Comment by RomainLanz** [comment]
Thanks!

**Comment by RomainLanz** [comment]
It looks like the generated feed has some issues: https://www.rssboard.org/rss-validator/check.cgi?url=https%3A%2F%2Fadonisjs.com%2F

Can you have a look at it? @msurdi

**Comment by msurdi** [comment]
Sure, I’ll check this over the weekend.

**Comment by msurdi** [comment]
@RomainLanz just opened https://github.com/adonisjs/adonisjs.com/pull/28 to address these issues.

**Pull request merged**